SFDigitalServices/sf-dahlia-web
# Dah 1452 cache multiple listing images

[1452](https://sfgovdt.jira.com/browse/DAH-1452)

Engineer Review Instructions

- Add a Listing Image to a listing in the full salesforce environment
- Go to PR environment (or localhost)
- Wait 10 minutes for the prefetch job to run. (we are running the prefetch job every ten minutes) Or if on local: Run the prefetch cache job following these instructions https://sfgovdt.jira.com/wiki/spaces/HOUS/pages/3468525649/Ruby+Debugging+Guide#Debugging-Ruby-Rake-jobs
- Once ran, go to the listing
- View the listing images, right click, and ensure the url is from cloudfront
- You can check the Postgres table and the S3 bucket as well if desired
Link to Postgres: https://data.heroku.com/dataclips/tctufwlnkivpmdbueefmsekcvorn
Link to S3 (might not have access): **https://s3.console.aws.amazon.com/s3/buckets/sf-gov-housing-dahlia-full?region=us-west-1&tab=objects**

More to come, ironing out a few things to be release ready
 - Need to add more logging
 - There is a long existing issue with our prefetch service where it does not correctly identify changed listings based on 'LastModifiedDate'. We can either use the prefetch:daily as our main job or change the 10 minute prefetch to always run the cache update. Realistically the jobs are doing the same thing. The main difference is the prefetch:daily runs a force=true to the salesforce cache.
 - Both of the rake jobs are relatively slow (has always been this way). May be opportunities to reduce unneeded calls to speed up execution time
 

Non-localhost testing ideas:
  - Either run the cache job directly from heroku? Not sure how to do this.
  - Or wait for the prefetch job to run - Runs at :01 every 10 minutes
  - Or change the frequency of prefetch daily in heroku for PR environment - our only options are 10 minute, hourly, or daily.
  
  Potential conflict between Photo_URL field caching and multiple listing images
   - investigating now, but long story short, we may need to add a boolean column to the listing_images table to indicate whether a listing_image record belongs to the Photo_URL field versus a listing image array item
